crates/ruff_linter/src/message/github.rs

```diff
@@ -2,13 +2,22 @@ use std::io::Write;
 
 use ruff_source_file::SourceLocation;
 
-use crate::fs::relativize_path;
+use crate::fs::{relativize_path, relativize_path_to};
 use crate::message::{Emitter, EmitterContext, Message};
 
 /// Generate error workflow command in GitHub Actions format.
 /// See: [GitHub documentation](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-error-message)
-#[derive(Default)]
-pub struct GithubEmitter;
+pub struct GithubEmitter {
+    project_dir: Option<String>,
+}
+
+impl Default for GithubEmitter {
+    fn default() -> Self {
+        Self {
+            project_dir: std::env::var("GITHUB_WORKSPACE").ok(),
```

Comment:
> The default working directory on the runner for steps, and the default location of your repository when using the [checkout](https://github.com/actions/checkout) action. For example, /home/runner/work/my-repo-name/my-repo-name.

I'm concerned that this will break users that clone their repository to a non-default location. 

It makes me wonder if we need to transform the paths twice:

1. Make the path absolute relative to the current working directory
2. Make the path relative to `GITHUB_WORKSPACE` 

Or are the paths already absolute?